utils/flink-components-testkit/src/main/scala/pl/touk/nussknacker/engine/flink/util/test/TestResultSinkFactory.scala

```diff
@@ -0,0 +1,73 @@
+package pl.touk.nussknacker.engine.flink.util.test
+
+import org.apache.flink.api.common.functions.FlatMapFunction
+import org.apache.flink.streaming.api.functions.sink.SinkFunction
+import pl.touk.nussknacker.engine.api._
+import pl.touk.nussknacker.engine.api.process.{Sink, SinkFactory}
+import pl.touk.nussknacker.engine.flink.api.process.{BasicFlinkSink, FlinkLazyParameterFunctionHelper}
+import pl.touk.nussknacker.engine.flink.util.test.TestResultSinkFactory.TestResultSink
+import pl.touk.nussknacker.engine.testmode.TestRunId
+
+import java.util.concurrent.ConcurrentSkipListMap
+
+class TestResultSinkFactory(runId: TestRunId) extends SinkFactory {
+
+  @MethodToInvoke
+  def create(@ParamName("value") value: LazyParameter[AnyRef]): Sink =
+    new TestResultSink(value, runId)
+
+}
+
+object TestResultSinkFactory {
+
+  private final case class SinkId(runId: TestRunId) extends Comparable[SinkId] {
+    override def compareTo(o: SinkId): Int = runId.id.compareTo(o.runId.id)
+  }
+
+  private[TestResultSinkFactory] val sinksOutputs = new ConcurrentSkipListMap[SinkId, Vector[AnyRef]]()
+
+  def extractSinkOutputFor(runId: TestRunId): Output = {
+    Option(sinksOutputs.remove(SinkId(runId)))
+      .map(l => Output.Present(l.toList))
+      .getOrElse(Output.None)
+  }
+
+  def clean(runId: TestRunId): Unit = {
```

Comment:
when is it called? isn't it unused?

Reply:
To be honest, I would prefer using `sinksOutputs.remove(runId)` during cleaning ;)